.github/workflows/test-code.yml

```diff
@@ -17,28 +17,17 @@ jobs:
         with:
           node-version: "22.5.1"
 
-      - name: Yarn update to v4
-        run: corepack enable && yarn set version 4.3.1
-
-      - name: Cache Yarn dependencies
-        id: yarn-cache
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
-        with:
-          path: |
-            ~/.cache/yarn
-            node_modules
-          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
```

Comment:
Could we keep this in and just change the paths it is caching.  Something like this:
```
      - name: Cache pnpm dependencies
        id: pnpm-cache
        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
        with:
          path: |
            node_modules
          key: ${{ runner.os }}-pnpm-${{ hashFiles('pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-
```

Reply:
I can try to put it back in. I was not sure how this would behave since PNPM does not save raw dependencies in `node_modules` but creates links to the PNPM store.

Reply:
Ah, in that case do not worry about it.  That's my misunderstanding about how PNPM works.  We can look at cache optimization later anyway.

Reply:
For reference: https://pnpm.io/motivation#saving-disk-space